### DISCUSSION TITLE
Widgets

### DISCUSSION THREAD

**Discussion by Pusnow** [MobiusSync/MobiusSync#118] [Ideas]
Currently, in order to perform a full sync or view the sync progress, it is necessary to open the app. However, the addition of widgets would enhance this functionality. For instance, a widget button could be implemented to initiate the sync process (buttons are available in iOS 17), or a progress circle within the widget could display the current sync progress. I'm not sure this is technically possible, this would provide a more convenient and efficient way to manage and monitor synchronization tasks.

**Comment by Tokarak** [comment]
My parralel idea in #119:
> I wanted to throw this idea in — but I can't say with certainty what it can or can't bring. My guess is that when a widget loads, that is sufficient to sync. This means that a sync can happen whenever the widget comes into view — eg when the phone is picked up. This can be useful for 'downsyncing' especially, since it will sync just before you open an app which depends on the freshest sync (e.g Logseq), without having to open the Möbius app first.

Some more thoughts:

- Minimum IOS target (Definitely can't be IOS 17)
- A sync could just require the widget to load (see ^^^)
- If there is a way to preload a widget (I know macos has a process which preloads the top 10 apps for faster load times), then the this would practically be background sync.
